python/main.py

```diff
@@ -1,42 +1,182 @@
 import os
 import logging
 import pathlib
+import json
+import sqlite3
+import hashlib
+from sqlite3 import Error
+
 from fastapi import FastAPI, Form, HTTPException
 from fastapi.responses import FileResponse
 from fastapi.middleware.cors import CORSMiddleware
 
+DB_PATH = './mercari.sqlite3'
+
 app = FastAPI()
 logger = logging.getLogger("uvicorn")
 logger.level = logging.INFO
-images = pathlib.Path(__file__).parent.resolve() / "image"
-origins = [ os.environ.get('FRONT_URL', 'http://localhost:3000') ]
+images = pathlib.Path(__file__).parent.resolve() / "images"
+origins = [os.environ.get('FRONT_URL', 'http://localhost:3000')]
+
 app.add_middleware(
     CORSMiddleware,
     allow_origins=origins,
     allow_credentials=False,
-    allow_methods=["GET","POST","PUT","DELETE"],
+    allow_methods=["GET", "POST", "PUT", "DELETE"],
     allow_headers=["*"],
 )
 
+# loading inventory
+with open('inventory.json') as js:
+    iv = json.load(js)
+
+# connect to db
+conn = sqlite3.connect(DB_PATH)
+c = conn.cursor()
+
+# create table
+c.execute("""CREATE TABLE IF NOT EXISTS items (
+                id INTEGER PRIMARY KEY,
+                name STRING,
+                category_id INTEGER,
+                image STRING,
+                FOREIGN KEY(category_id) REFERENCES category(category_id)
+          )""")
+
+c.execute("""CREATE TABLE IF NOT EXISTS category (
+               category_id INTEGER PRIMARY KEY,
+               name STRING UNIQUE
+          )""")
+
+conn.commit()
+conn.close()
+
+
+# Hash the image
+def hash_image(filepath):
+    with open(filepath, "rb") as f:
+        bts = f.read()  # read entire file as bytes
+        readable_hash = hashlib.sha256(bts).hexdigest()
+        return f'{readable_hash}.jpg'
+
+
+def get_specific_items(**args):
+    try:
+        conn = sqlite3.connect(DB_PATH)
+        c = conn.cursor()
+        sql = """SELECT items.name,
+             items.image, 
+             category.name as category_name
+             FROM items INNER JOIN category
+             ON items.category_id =category.category_id"""
+        c.execute(sql)
```

Comment:
sql string -> execute -> append another conditional string -> execute again.
why do we need to execute twice?